lib/sprite_batch.dart

```diff
@@ -29,16 +29,64 @@ class SpriteBatch {
 
   Vector2 get size => Vector2Extension.fromInts(width, height);
 
+  /// Add a new sprite using a RSTransform.
+  ///
+  /// The [rect] parameter is the source location on the [atlas]. You can position it
+  /// on the canvas using the [offset] parameter.
+  ///
+  /// The [color] paramater allows you to render a color behind the sprite, as a background color.
+  ///
+  /// The [add] method may be a simpler way to add a sprite to the batch. However,
+  /// if there is a way to factor out the computations of the sine and cosine of the
+  /// rotation so that they can be reused over multiple calls to this constructor,
+  /// it may be more efficient to directly use this method instead.
   void addTransform({
     @required Rect rect,
     RSTransform transform,
     Color color,
   }) {
+    transform ??= defaultTransform;
     rects.add(rect);
-    transforms.add(transform ?? defaultTransform);
+    transforms.add(transform);
     colors.add(color ?? defaultColor);
+
+    /// Because `drawAtlas` is not supported on the web we also
+    /// build a `Matrix4` based on the given `RSTransform`.
+    matrices.add(Matrix4(
```

Comment:
Any reason for not doing that only on web?

Reply:
There is no computation taking place so adding an extra if statement would "impact" the performance more than what we would benefit from it IMHO, and this also ensures those lists will never be out of sync.

Reply:
I am wondering now which one would impact more, an `if` or instantiating something, but the last statement convinced me already.

Reply:
Well an `if` would have a lesser impact I guess, but it would be extra for the web then :-)